src/cmd-buildextend-metal

```diff
@@ -227,7 +232,19 @@ runvm_with_cache_snapshot "$snapshot" -- /usr/lib/coreos-assembler/runvm-osbuild
 
 if [[ "${image_type}" == "qemu-secex" ]]; then
     if [ ! -f "${genprotimgvm}" ]; then
-        fatal "No genprotimgvm provided at ${genprotimgvm}"
+        if [ ! -f "${hostkey}" ]; then
+            fatal "No hostkey and no genprotimgvm provided"
+        fi
+        echo "Generating genprotimgvm locally"
+        ignition=$(mktemp -p "${tmp_builddir}")
+        butane -p -d "$(dirname "${hostkey}")" /usr/lib/coreos-assembler/secex-genprotimgvm-scripts/genprotimg.bu -o "${ignition}"
+
+        genprotimgvm=/srv/fake-secure-vm.qcow2
```

Comment:
Hmm, this should probably be below the cosa workdir. How about putting it in `${workdir}/tmp/`?